### PR TITLE
Added query, error formalization, and better tests + some fixes

### PR DESCRIPTION
**Pull request by jonasfj**
Can you approve this... Most of it is test refactoring feel free to ignore that...

Notably this fixes a few minor bugs, and adds support for query string parameters and sets `err.code` when an error code is available in response... It will be once error formalization is deployed...

### CODE REVIEW

**Comment by jonasfj** [comment]
@petemoore (and @jhford) we need to add similar support for query-string options in python and go-lang clients...

Notice that query string options are **always optional**. They are intended for things like continuation tokens...

----
Unrelated I had to rerun circle ci because of some git issue... maybe we should try to move this back to travis... :)

**Comment by petemoore** [comment]
I added support in the go client a while back, so nothing to do there, I believe.

File: lib/client.js [hunk 1]
```diff
@@ -59,7 +60,15 @@ var _defaultOptions = {
 };
 
 /** Make a request for a Client instance */
-var makeRequest = function(client, method, url, payload) {
+var makeRequest = function(client, method, url, payload, query) {
+  // Add query to url if present
+  if (query) {
+    query = querystring.stringify(query);
+    if (query.length > 0) {
+      url += '?' + query;
```
Comment (jhford) [inline lib/client.js]: We have url.format, can we use that instead?

**Comment by jhford** [comment]
Can you add something to the Readme.md before merging?  The python client will need to be update to fix that.  Are we using query string options yet?

File: lib/client.js (hunk 1 above)
Comment (jonasfj) [inline lib/client.js]: Then I have to parse the urls...

**Comment by jonasfj** [comment]
> Are we using query string options yet?

Nope.... We will be soon... both index and queue will need this for `continuationToken` when listing namespaces and taskGroups respectively.

**Comment by jonasfj** [comment]
Fixed tests and moved to travis, getting rid of circle-ci

**Pull request merged**